exe/daemon/main.go

```diff
@@ -17,16 +19,24 @@ func main() {
 	flag.Parse()
 
 	if err := logging.SetLogLevel("daemon", "debug"); err != nil {
-		panic(err)
+		log.Fatal(err)
 	}
 
-	_, cancel, ds, h, dht, api := util.Build(*repo, *port, *proxyAddr, true)
+	_, cancel, ds, h, dht, ts := util.Build(*repo, *port, *proxyAddr, true)
 
 	defer cancel()
 	defer dht.Close()
-	defer api.Close()
+	defer ts.Close()
 	defer ds.Close()
 
+	server, err := api.NewServer(context.Background(), ds, ts, api.Options{
```

Comment:
Same datastore for possibly multiple stores. I think this won't work correctly. (See above 'Reminder:' comment)
Maybe we could make the store ID a first class citizen and use it as a prefix for using the datastore. This would allow sharing a single datastore for multiple Stores.

Reply:
Other option is using [keytransform](https://github.com/ipfs/go-datastore/tree/master/keytransform) to build a wrapper with the ID prefix, and make it transparent for the Store.

Reply:
OK cool. I was definitely confused about that. I think we'll want to leverage the same repo across stores if at all possible. I'm imagining a client hub can be serving many stores. I can take a crack at a manager.